### PR TITLE
16585 update smallrye

### LINKED ISSUE #16585: Update SmallRye Config to version latest
There are various issues fixed in 2.2.0. One that we raised was https://github.com/smallrye/smallrye-config/issues/522

### PR DESCRIPTION
**Pull request by benjamin-confino**
fixes #16585

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Review by tevans78 — APPROVED** [review]

**Comment by benjamin-confino** [comment]
#build

**Review by Azquelt — APPROVED** [review]

File: dev/io.openliberty.microprofile.config.2.0.internal_fat/fat/src/io/openliberty/microprofile/config/internal_fat/apps/TestUtils.java [hunk 1]
```diff
@@ -41,6 +41,18 @@ public static void assertContains(Config config, Map<String, String> props) {
...
+        if (searchFor == null) {
+            throw new IllegalArgumentException("You cannot search for a null");
+        }
+        if (searchWithin == null ) {
+            throw new AssertionError("Attempting to look for string \"" + searchFor + "\" but the string to search within was null");
+        }
+        if (searchWithin != null && !searchWithin.contains(searchFor)) {
+            throw new AssertionError("\"" + searchWithin + "\" does not contain \"" + searchFor + "\"");
+        }
+    }
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.config.2.0.internal_fat/fat/src/io/openliberty/microprofile/config/internal_fat/apps/TestUtils.java:54]: This is the same as `assertThat(searchWithin, containsString(searchFor))`

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Pull request merged**